hal/src/sercom/v2/uart/reg.rs

```diff
@@ -0,0 +1,400 @@
+//! Register-level access to UART configuration
+
+use super::{BaudMode, BitOrder, Parity, StopBits};
+
+use crate::sercom::v2::*;
+use crate::target_device as pac;
+
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+use pac::sercom0::usart::ctrla::MODE_A;
+
+#[cfg(feature = "min-samd51g")]
+use pac::sercom0::usart_int::ctrla::MODE_A;
+
+use crate::time::Hertz;
+
+pub struct Registers<S: Sercom> {
+    sercom: S,
+}
```

Comment:
Since you've removed interior mutability from the PAC `SERCOM` struct, you can `unsafe impl Sync for Registers`, even though the PAC struct isn't `Sync`.